src/dsm/headers/SparseMatrix.hpp

```diff
@@ -371,6 +377,22 @@ namespace dsm {
     _matrix = new_matrix;
   }
 
+  template <typename Index, typename T>
+    requires std::unsigned_integral<Index>
+  void SparseMatrix<Index, T>::emptyRow(Index index) {
+    for (Index i = 0; i < _cols; ++i) {
```

Comment:
Isn't it better to iterate over the row elements?
e.g. iterating getCol and erasing it from the matrix

Reply:
If you iterate over getRow you iterate over a copy of the matrix, not the real one. If you erase elements you are only erasing them from the copy.

Reply:
Might create a new matrix method for this